packages/connect/src/device/Device.ts

```
@@ -527,6 +528,76 @@ export class Device extends TypedEmitter<DeviceEvents> {
     async getFeatures() {
         const { message } = await this.getCommands().typedCall('GetFeatures', 'Features', {});
         this._updateFeatures(message);
+
+        if (
+            !this.features.language_version_matches &&
+            this.features.language &&
+            this.atLeast('2.7.0')
+        ) {
+            _log.info('language version mismatch. silently updating...');
+            await this.changeLanguage({ language: this.features.language });
+        }
+    }
+
+    async changeLanguage({
+        language,
+        binary,
+    }: { language?: undefined; binary: ArrayBuffer } | { language: string; binary?: undefined }) {
+        if (language === 'en-US') {
+            return this._uploadTranslationData(null);
+        }
+
+        if (binary) {
+            return this._uploadTranslationData(binary);
+        }
+
+        const downloadedBinary = await getLanguage({
```

silent error when offline